File: apps/mobile/lib/features/polls/domain/entities/poll.dart
import 'package:equatable/equatable.dart';
import 'package:anketa/features/auth/domain/entities/user.dart';
import 'option.dart';

class Poll extends Equatable {
  final String id;
  final String question;
  final String? description;
  final String? category;
  final DateTime? expiresAt;
  final DateTime createdAt;
  final User author;
  final List<Option> options;
  final int totalVotes;
  final int commentsCount;
  final bool hasVoted;
  final String? userVotedOptionId;

  const Poll({
    required this.id,
    required this.question,
    this.description,
    this.category,
    this.expiresAt,
    required this.createdAt,
    required this.author,
    required this.options,
    this.totalVotes = 0,
    this.commentsCount = 0,
    this.hasVoted = false,
    this.userVotedOptionId,
  });

  bool get isExpired =>
      expiresAt != null && expiresAt!.isBefore(DateTime.now());

  @override
  List<Object?> get props => [
        id,
        question,
        description,
        category,
        expiresAt,
        createdAt,
        author,
        options,
        totalVotes,
        commentsCount,
        hasVoted,
        userVotedOptionId,
      ];
}

File: packages/shared/src/types/poll.ts
import type { UserDto } from './user';

export interface OptionDto {
  id: string;
  text: string;
  imageUrl: string | null;
  votesCount: number;
  percentage: number;
}

export interface PollDto {
  id: string;
  question: string;
  description: string | null;
  category: string | null;
  expiresAt: string | null;
  totalVotes: number;
  commentsCount: number;
  hasVoted: boolean;
  userVotedOptionId: string | null;
  author: UserDto;
  options: OptionDto[];
  createdAt: string;
}

export interface PollListDto {
  data: PollDto[];
  meta: PaginationMeta;
}

export interface PaginationMeta {
  total: number;
  page: number;
  limit: number;
  hasNextPage: boolean;
}

export interface CreatePollOptionDto {
  text: string;
  imageUrl?: string;
}

export interface CreatePollDto {
  question: string;
  description?: string;
  category?: string;
  expiresAt?: string;
  options: CreatePollOptionDto[];
}
